hazelcast/src/test/java/com/hazelcast/map/BasicMapTest.java

```diff
@@ -742,8 +748,17 @@ public void run() {
         assertOpenEventually(waitTryPutFailure);
 
         map.unlock("key1");
-        assertOpenEventually(latch);
+        try {
+            boolean completed = latch.await(ASSERT_TRUE_EVENTUALLY_TIMEOUT, TimeUnit.SECONDS);
+            assertTrue("tryPut operations were not completed in time, counter was " + counter.get() + " and failure"
```

Comment:
Unwrapped `assertOpenEventually` to evaluate `counter.get()` and `failureMessageBuilder.toString()` at time of (potential) failure.